### DISCUSSION TITLE
[Raspberry Pi] Not able to launch QML interface

### DISCUSSION THREAD

**Discussion by shoulih** [cagnulein/qdomyos-zwift#266] [Q&A]
Looks like some qt packages might be missing on Raspberry Pi when launching QML interface.
pi@raspberrypi:~ $ sudo ./qdomyos-zwift/src/qdomyos-zwift -qml

Debug: main.cpp int main(int, char**) version  ""
Debug:   Bluez 5 detected.
Debug:   Creating QtBluezDiscoveryManager
Debug:   Discovered:  "C1:1F:39:A7:6A:C5" "I_EB" Num UUIDs 5 total device 0 cached RSSI 0 Class 0
Debug: bluetooth.cpp void bluetooth::debug(QString) "Mon Mar 8 11:54:26 2021 1615233266645 Found new device: I_EB (C1:1F:39:A7:6A:C5) \u0000:\u0000"
Debug:   void QBluetoothDeviceDiscoveryAgentPrivate::stop()
Debug: bluetooth.cpp void bluetooth::debug(QString) "Mon Mar 8 11:54:26 2021 1615233266667 BTLE scanning stops"
Debug: bluetooth.cpp void bluetooth::debug(QString) "Mon Mar 8 11:54:26 2021 1615233266673 Found new device: I_EB (C1:1F:39:A7:6A:C5)"
Debug:   Detecting bluetoothd version
Debug:   Using /proc/<pid>/exe
Debug:   Detected bluetoothd version 5.50
Debug:   Bluetoothd: "5.50"
Warning:   Using BlueZ LE DBus API
Debug:   QLowEnergyControllerPrivateBluezDBus::connectToDevice()
Debug: proformbike.cpp void proformbike::controllerStateChanged(QLowEnergyController::ControllerState) controllerStateChanged QLowEnergyController::ControllerState(ConnectingState)
Warning:   QQmlApplicationEngine failed to load component
Warning:   qrc:/main.qml:2 module "QtQuick.Controls" is not installed
qrc:/main.qml:1 module "QtQuick" is not installed
qrc:/main.qml:4 module "QtQuick.Dialogs" is not installed
qrc:/main.qml:3 module "QtQuick.Controls.Material" is not installed
qrc:/main.qml:2 module "QtQuick.Controls" is not installed
qrc:/main.qml:1 module "QtQuick" is not installed
qrc:/main.qml:4 module "QtQuick.Dialogs" is not installed
qrc:/main.qml:3 module "QtQuick.Controls.Material" is not installed
...

Aborted

**Comment by cagnulein** [comment]
Install qml-module*

Il lun 8 mar 2021, 21:14 shoulih <notifications@github.com> ha scritto:

**Comment by shoulih** [comment]
Still got the following:
...
Warning:   QQmlApplicationEngine failed to load component
QQmlApplicationEngine failed to load component
Warning:   qrc:/main.qml:2 module "QtQuick.Controls" version 2.5 is not installed

qrc:/main.qml:2 module "QtQuick.Controls" version 2.5 is not installed

Aborted

**Comment by shoulih** [comment]
Now it is asking for "QtQuick" version 2.12 after I did an "apt-get update"

Warning:   QQmlApplicationEngine failed to load component
QQmlApplicationEngine failed to load component
Warning:   qrc:/main.qml:1 module "QtQuick" version 2.12 is not installed

qrc:/main.qml:1 module "QtQuick" version 2.12 is not installed

Aborted

**Comment by cagnulein** [comment]
Did you do this also

sudo apt install qml-module-qtquick-controls2

The last time I tried it works on the raspberry.
Anyway i guess i can add the same version in all the module.
If you want you could change it and tell me if it works


Il mar 9 mar 2021, 03:42 shoulih <notifications@github.com> ha scritto:

**Comment by shoulih** [comment]
Yes.

pi@raspberrypi:~ $ sudo apt install qml-module-qtquick-controls2
Reading package lists... Done
Building dependency tree       
Reading state information... Done
qml-module-qtquick-controls2 is already the newest version (5.11.3+dfsg-2).
0 upgraded, 0 newly installed, 0 to remove and 57 not upgraded.

Please advise.

Thanks,

**Comment by cagnulein** [comment]
change this line qrc:/main.qml:1 module "QtQuick" version 2.12 is not installed

to use the version 2.12

**Comment by shoulih** [comment]
not quite follow what to change:

Below are the imports in "main.qml" file:

import QtQuick 2.12
import QtQuick.Controls 2.5
import QtQuick.Controls.Material 2.12
import QtQuick.Dialogs 1.0

Do we want me to try a different version of imports?

**Comment by cagnulein** [comment]
Yes try to use the one that you have installed

Il mar 9 mar 2021, 16:58 shoulih <notifications@github.com> ha scritto:

**Comment by shoulih** [comment]
now it is saying: qrc:/main.qml:2 module "QtQuick.Controls" version 2.5 is not installed
and below shows what I installed on the pi:
pi@raspberrypi:~/qdomyos-zwift/src $ dpkg -l | grep Controls
ii  libqt5quickcontrols2-5:armhf                  5.11.3+dfsg-2                       armhf        Qt 5 Quick Controls 2 library
ii  qml-module-org-kde-qqc2desktopstyle           5.54.0-1                            armhf        Qt Quick Controls 2: Desktop Style
ii  qml-module-qtquick-controls:armhf             5.11.3-2                            armhf        Qt 5 Quick Controls QML module
ii  qml-module-qtquick-controls2:armhf            5.11.3+dfsg-2                       armhf        Qt 5 Qt Quick Controls 2 QML module

any suggestion on what QtQuick.Controls version I need to change to in the main.qml file:

import QtQuick 2.11
import QtQuick.Controls 2.5
import QtQuick.Controls.Material 2.11
import QtQuick.Dialogs 1.0

**Comment by cagnulein** [comment]
Right now i don't have the time to investigate it.
I saw some post on stack overflow with the same topic.
You should try to play with it and let me know.
Thanks and sorry :(

Il mar 9 mar 2021, 18:24 shoulih <notifications@github.com> ha scritto:

**Comment by shoulih** [comment]
sure, will give you an update on that.  Thanks again!

**Comment by shoulih** [comment]
updated version numbers of the imports in the *.qml files on my pi and able to get QML interface now!
https://doc.qt.io/qt-5/qtquickcontrols-index.html

Thanks again!

**Comment by cagnulein** [comment]
Great!
Which file did you change?
Could you please upload them here?
Thanks

Il mer 10 mar 2021, 01:03 shoulih <notifications@github.com> ha scritto:

**Comment by shoulih** [comment]
I made the version update on all qml files and please find them attached here.
[qlm_files.zip](https://github.com/cagnulein/qdomyos-zwift/files/6117356/qlm_files.zip)

**Comment by cagnulein** [comment]
ok i will check them when i will upload the next production one on the store. thanks

**Comment by shoulih** [comment]
cloned the latest code from github and ran into compilation error below on raspberry pi:

![image](https://user-images.githubusercontent.com/66762397/114497855-d5ab9d80-9bd7-11eb-86fb-c0d8aadb0c71.png)

can you please look into it when you get a chance?

Thanks,

**Comment by cagnulein** [comment]
Check the updated instructions, you need to get the submodule too now

Il mar 13 apr 2021, 06:42 shoulih ***@***.***> ha scritto:
